src/io.js

```diff
@@ -146,3 +145,93 @@ export default function twoline2satrec(longstr1, longstr2) {
 
   return satrec;
 }
+
+/* -----------------------------------------------------------------------------
+ *
+ *                           function json2satrec
+ *
+ *  this function converts the OMM json data to variables and initializes the sgp4 
+ *    variables. several intermediate varaibles and quantities are determined. note 
+ *    that the result is a structure so multiple satellites can be processed 
+ *    simultaneously without having to reinitialize. the verification mode is an 
+ *    important option that permits quick checks of any changes to the underlying 
+ *    technical theory. this option works using a modified tle file in which the 
+ *    start, stop, and delta time values are included at the end of the second line
+ *    of data. this only works with the verification mode. the catalog mode simply 
+ *    propagates from -1440 to 1440 min from epoch and is useful when performing 
+ *    entire catalog runs.
+ *
+ *  author        : Hariharan Vitaladevuni                   18 Aug 2023
+ *                  Theodore Kruczek                         19 Aug 2023
+ *
+ *  inputs        :
+ *    jsonobj     - OMM json data
+ *    opsmode     - mode of operation afspc or improved 'a', 'i'. Default: 'i'.
+ *
+ *  outputs       :
+ *    satrec      - structure containing all the sgp4 satellite information
+ *
+ *  coupling      :
+ *    days2mdhms  - conversion of days to month, day, hour, minute, second
+ *    jday        - convert day month year hour minute second into julian date
+ *    sgp4init    - initialize the sgp4 variables
+ * 
+ *  warning       : the epoch date in OMM format is more accurate than TLE format!
+ *                  this will result in extremely close, but different 
+ *                  position/velocity values. Depending on your use case, it may
+ *                  be better to use twoline2satrec, but for the average user this
+ *                  will provide comparable results.
+ *
+ *  references    :
+ *    https://celestrak.org/NORAD/documentation/gp-data-formats.php
+ --------------------------------------------------------------------------- */
+export function json2satrec(jsonobj, opsmode = 'i') {
+  const satrec = {};
+  satrec.error = 0;
+
+  satrec.satnum = jsonobj.NORAD_CAT_ID.toString().padStart(5, '0');
```

Comment:
Do we really need to pad this? TLE from Space-Track don't pad NORAD id with zeroes, for example.

Reply:
Given that we are now using 6 digit ids it is moot. Space-Track does in some cases and doesn't in other cases. They are inconsistent with their padding. Celestrak pads all ids with 0s. I was focusing on consistency, but you can remove it.

Reply:
I'd prefer keeping that as originally provided in JSON, especially since it might be 5 digits at the time of PR, 6 digits now, 7 digits in a few more years etc. I think consistency is not in the scope of the library, but for the users if they want that.